controllers/provider-azure/pkg/webhook/controlplane/ensurer.go

```diff
@@ -143,3 +146,27 @@ func (e *ensurer) EnsureKubeletConfiguration(ctx context.Context, kubeletConfig
 	delete(kubeletConfig.FeatureGates, "CSIDriverRegistry")
 	return nil
 }
+
+//ShouldProvisionKubeletCloudProviderConfig returns if the cloudprovider.config file should be added to the kubelet configuration.
+func (e *ensurer) ShouldProvisionKubeletCloudProviderConfig() bool {
+	return true
+}
+
+//EnsureKubeletCloudProviderConfig ensures that the cloudprovider.config file conforms to the provider requirements.
+func (e *ensurer) EnsureKubeletCloudProviderConfig(ctx context.Context, data *string, namespace string) error {
+	// Get `cloud-provider-config` ConfigMap
+	var cm corev1.ConfigMap
+	err := e.client.Get(ctx, kutil.Key(namespace, openstack.CloudProviderConfigName), &cm)
+	if err != nil {
+		return errors.Wrapf(err, "could not get configmap with name '%s' and namespace '%s'", openstack.CloudProviderConfigName, namespace)
+	}
+
+	// Check if the `cloud-provider-config` has "cloudprovider.conf" key
+	if cm.Data == nil || cm.Data["cloudprovider.conf"] == "" {
```

Comment:
Let's make `cloudprovider.conf` a constant somewhere

Reply:
There is already a constant in gardener - gardener/pkg/operation/common/types.go. I will use that one.

Reply:
This constant will be removed as part of @stoyanr's PR in Gardener as Gardener itself will not know anything about a cloud provider config anymore. Let's define it in this repo, ideally it should be defined in every provider implementation itself (as every provider can independently decide how it wants to name this data key).